### PR TITLE
feat(keychain)!: expose clear method instead of clearing in plugin

### PR DESCRIPTION
**Pull request by diegomura**
Let's make keychain application lifecycle agnostic. Hooks are just used for clearing keychain, so let's just expose a clear method `wallet` module can call

### CODE REVIEW

File: features/keychain/module/memoized-keychain.js [hunk 1]
```diff
@@ -49,12 +50,15 @@ class MemoizedKeychain extends Keychain {
   }
 
   clone = () => new MemoizedKeychain(this.#cloneOpts)
+
+  clear = async () => this.#storage.delete(CACHE_KEY)
```
Comment (mvayngrib) [inline features/keychain/module/memoized-keychain.js:54]: given MemoizedKeychain extends Keychain, should we also call super.clear()?

**Review by mvayngrib — DISMISSED** [review]
utACK

this seems like a breaking change, given we don't clear storage here anymore 🤔 . Wdyt?

**Comment by mvayngrib** [comment]
@diegomura some lint errors

**Comment by mvayngrib** [comment]
@diegomura can u rebase? lint errors were on master, just fixed

**Review by kewde — APPROVED** [review]
utACK

**Pull request merged**